src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php

```diff
@@ -10,8 +14,82 @@
  */
 class DashboardController extends Controller
 {
+
+    /**
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
     public function indexAction()
     {
-        return $this->render('DashboardHubAppBundle:Dashboard:index.html.twig');
+        return $this->render(
+                'DashboardHubAppBundle:Dashboard:index.html.twig',
+                array(
+                    'dashboards' => $this->get('dashboardhub_app_main.service.dashboard')
+                        ->findAllByAuthenticatedUserAndDefaults()
+                )
+            );
+    }
+
+    public function addAction(Request $request)
+    {
+        $dashboard = new Dashboard();
+        $form = $this->createForm(new DashboardType(), $dashboard);
+
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $dashboard = $form->getData();
+
+            $this->get('dashboardhub_app_main.service.dashboard')
+                ->save($dashboard);
+
+            $request->getSession()->getFlashBag()->add(
```

Comment:
Ok, I'll merge this then first
On 09 Mar 2015 4:39 AM, "Eddie Abou-Jaoude" <notifications@github.com>
wrote:

> In src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php
> <https://github.com/DashboardHub/PipelineDashboard/pull/60#discussion_r26015117>
> :
>
> > +    }
> > +
> > +    public function addAction(Request $request)
> > +    {
> > +        $dashboard = new Dashboard();
> > +        $form = $this->createForm(new DashboardType(), $dashboard);
> > +
> > +        $form->handleRequest($request);
> > +
> > +        if ($form->isValid()) {
> > +            $dashboard = $form->getData();
> > +
> > +            $this->get('dashboardhub_app_main.service.dashboard')
> > +                ->save($dashboard);
> > +
> > +            $request->getSession()->getFlashBag()->add(
>
> The next PR contains a refactor, so made sense to do it in that (e.g. form
> to services). Flash message still required.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DashboardHub/PipelineDashboard/pull/60/files#r26015117>
> .
>